server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -85,7 +86,12 @@ public WriteableSetting(StreamInput in) throws IOException {
         if (hasFallback) {
             fallback = new WriteableSetting(in);
         }
-        // We are using known types so don't need the parser
+        // Read the parser
+        WriteableSetting parser = null;
+        boolean isParserWriteable = in.readBoolean();
+        if (isParserWriteable) {
+            parser = new WriteableSetting(in);
```

Comment:
Here you'll need a switch/case to read in an `IntegerParser`, `DoubleParser`, etc.  Should be a similar implementation to the `readDefaultValue()` you call on line 82.

It surprises me that this doesn't fail one of the tests in `WriteableSettingTests` since we wrote bytes for a parser but are reading in bytes for a setting.... unless we also wrote a setting....